Errors with `dockerfile` parser

Hi.
I updated to `0.17.1` and trying to use new `dockerfile` parser to format dockerfile.
But it dies with error:

```
panic: runtime error: index out of range
RuntimeError: unreachable
    at main.runtime.runtimePanicAt (wasm://wasm/main-004aad7a:wasm-function[211]:0x10867)
    at main.runtime.lookupPanic (wasm://wasm/main-004aad7a:wasm-function[18]:0x1dff)
    at main.github.com/reteps/dockerfmt/lib.formatRun (wasm://wasm/main-004aad7a:wasm-function[1788]:0xc012b)
    at main.github.com/reteps/dockerfmt/lib.FormatNode (wasm://wasm/main-004aad7a:wasm-function[1780]:0xbe2c2)
    at main.(*github.com/reteps/dockerfmt/lib.ParseState).processNode (wasm://wasm/main-004aad7a:wasm-function[1778]:0xbde86)
    at main.(*github.com/reteps/dockerfmt/lib.ParseState).processNode (wasm://wasm/main-004aad7a:wasm-function[1778]:0xbdf31)
    at main.formatBytes.command_export (wasm://wasm/main-004aad7a:wasm-function[1797]:0xc6475)
    at formatDockerfileContents (file:///d:/projects/softvisio-node/cli/node_modules/@reteps/dockerfmt/dist/format.js:20:27)
    at async printAstToDoc (file:///d:/projects/softvisio-node/cli/node_modules/prettier/index.mjs:20124:16)
    at async coreFormat (file:///d:/projects/softvisio-node/cli/node_modules/prettier/index.mjs:20538:14)
```
`Dockerfile`:
```
RUN \
    # cleanup
    ls -la
```

Seems, that such comments syntax is not supported.

--- a/.changeset/rotten-dancers-bow.md
+++ b/.changeset/rotten-dancers-bow.md
@@ -1,0 +1,5 @@
+---
+"prettier-plugin-sh": patch
+---
+
+fix: fallback to `sh` printer when `dockerfmt` throws

--- a/packages/sh/src/index.ts
+++ b/packages/sh/src/index.ts
@@ -113,18 +113,62 @@ const dockerPrinter: Printer<string> = {
   async print(
     path,
     {
+      filepath,
+
+      // parser options
+      keepComments = true,
+      variant,
+      stopAt,
+      recoverErrors,
+
+      // printer options
       useTabs,
       tabWidth,
-      indent = useTabs ? 0 : tabWidth,
-      spaceRedirects = false,
-    }: DockerfilePrintOptions,
+      indent = useTabs ? 0 : (tabWidth ?? 2),
+      binaryNextLine = true,
+      switchCaseIndent = true,
+      spaceRedirects,
+      // eslint-disable-next-line sonarjs/deprecation
+      keepPadding,
+      minify,
+      singleLine,
+      functionNextLine,
+    }: ShPrintOptions,
   ) {
     const formatDockerfileContents = await getFormatDockerfileContents()
-    return formatDockerfileContents(path.node, {
-      indent,
-      spaceRedirects,
-      trailingNewline: true,
-    })
+    try {
+      return await formatDockerfileContents(path.node, {
+        indent,
+        spaceRedirects: spaceRedirects ?? false,
+        trailingNewline: true,
+      })
+    } catch {
+      /*
+       * `dockerfmt` is buggy now and could throw unexpectedly, so we fallback to
+       * the `sh` printer automatically in this case.
+       *
+       * @see {https://github.com/reteps/dockerfmt/issues/21}
+       * @see {https://github.com/reteps/dockerfmt/issues/25}
+       */
+      return processor(path.node, {
+        print: true,
+        filepath,
+        keepComments,
+        variant,
+        stopAt,
+        recoverErrors,
+        useTabs,
+        tabWidth,
+        indent,
+        binaryNextLine,
+        switchCaseIndent,
+        spaceRedirects: spaceRedirects ?? true,
+        keepPadding,
+        minify,
+        singleLine,
+        functionNextLine,
+      })
+    }
   },
 }
 

--- a/packages/sh/test/fixtures/445.Dockerfile
+++ b/packages/sh/test/fixtures/445.Dockerfile
@@ -1,0 +1,7 @@
+FROM nginx
+WORKDIR /app
+ARG PROJECT_DIR=/
+ARG NGINX_CONF=nginx.conf
+COPY $NGINX_CONF /etc/nginx/conf.d/nginx.conf
+COPY $PROJECT_DIR /app
+CMD mkdir --parents /var/log/nginx && nginx -g "daemon off;"

